packages/editor/src/components/document-bar/index.js

```diff
@@ -48,40 +48,46 @@ const icons = {
 };
 
 export default function DocumentBar() {
-	const { isEditingTemplate, templateId, postType, postId } = useSelect(
-		( select ) => {
+	const { isEditingTemplate, templateId, postType, postId, goBack } =
+		useSelect( ( select ) => {
 			const {
 				getRenderingMode,
 				getCurrentTemplateId,
 				getCurrentPostId,
 				getCurrentPostType,
+				getEditorSettings,
 			} = select( editorStore );
 			const _templateId = getCurrentTemplateId();
+			const back = getEditorSettings().goBack;
 			return {
 				isEditingTemplate:
 					!! _templateId && getRenderingMode() === 'template-only',
 				templateId: _templateId,
 				postType: getCurrentPostType(),
 				postId: getCurrentPostId(),
+				goBack: typeof back === 'function' ? back : undefined,
 			};
-		},
-		[]
-	);
+		}, [] );
 	const { getEditorSettings } = useSelect( editorStore );
 	const { setRenderingMode } = useDispatch( editorStore );
 
+	const handleOnBack = () => {
+		if ( isEditingTemplate ) {
```

Comment:
I was talking about "template-only" mode because in my mind, this would just be `goBack` like the other one. There would be no special case for templates.

Reply:
We were planning to follow up with an additional PR straight after this one to bring the template mode into line with this new approach otherwise this PR gets a bit polluted with changes not related to the pattern implementation.